src/map.cpp

```diff
@@ -1032,69 +1032,82 @@ void Map::removeNodeTimer(v3s16 p)
 	block->m_node_timers.remove(p_rel);
 }
 
-bool Map::isOccluded(v3s16 p0, v3s16 p1, float step, float stepfac,
-		float start_off, float end_off, u32 needed_count)
-{
-	float d0 = (float)BS * p0.getDistanceFrom(p1);
-	v3s16 u0 = p1 - p0;
-	v3f uf = v3f(u0.X, u0.Y, u0.Z) * BS;
-	uf.normalize();
-	v3f p0f = v3f(p0.X, p0.Y, p0.Z) * BS;
+bool Map::isOccluded(v3s16 pos_camera, v3s16 pos_target, float step,
+	float stepfac, float offset, float end_offset, u32 needed_count)
+{
+	v3f direction = intToFloat(pos_target - pos_camera, BS);
+	float distance = direction.getLength();
+	if (distance > 0.0f)
+		direction /= distance; // Normalize vector
+
+	v3f pos_origin_f = intToFloat(pos_camera, BS);
 	u32 count = 0;
-	for(float s=start_off; s<d0+end_off; s+=step){
-		v3f pf = p0f + uf * s;
-		v3s16 p = floatToInt(pf, BS);
-		MapNode n = getNode(p);
-		const ContentFeatures &f = m_nodedef->get(n);
-		if(f.drawtype == NDT_NORMAL){
-			// not transparent, see ContentFeature::updateTextures
+	bool is_valid_position;
+
+	for (; offset < distance + end_offset; offset += step){
+		v3f pos_node_f = pos_origin_f + direction * offset;
+		MapNode node = getNode(floatToInt(pos_node_f, BS), &is_valid_position);
+
+		if (is_valid_position &&
+					!m_nodedef->get(node).light_propagates) {
```

Comment:
2 indents?

Reply:
>When breaking conditionals, indent following lines of the conditional with **two** tabs and the statement body with one tab.

https://dev.minetest.net/Code_style_guidelines#Spaces

But here are 3 tabs, which is wrong.